server/services/stream.py

```
@@ -4,42 +4,75 @@
-        # 이미 종료된 프로세스는 정리함
+        # 이미 종료된 프로세스는 정리함 (로그 핸들 포함)
         del _processes[key]
+        _close_log(key)
+
+    # 로그 파일 준비함 — 디렉토리 보장 후 덮어쓰기 모드로 open함
+    _LOG_DIR.mkdir(exist_ok=True)
+    log_path = _LOG_DIR / f"core_{group_id}_{subject_index}.log"
+    log_file = open(log_path, "w", encoding="utf-8")
 
+    # python -u 로 자식 출력 버퍼링 제거 → 실시간 로그 확보함 (D7)
     proc = subprocess.Popen(
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)